.circleci/chain_on_pr.sh

```diff
@@ -1,102 +1,39 @@
-#!/bin/bash -e
+#!/bin/bash
 
-export NVM_DIR="$HOME/.nvm"
-[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
-[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
+echo "Downloading the current testnet Boyar config.json"
+curl -O $BOOTSTRAP_URL
+echo "Done downloading! Let's begin by cleaning up the testnet of any stale networks for PRs that are already closed"
 
-nvm use $NODE_VERSION
+node .circleci/testnet-cleanup-mark.js
 
-cd .circleci && npm install @orbs-network/orbs-nebula && cd ..
+echo "Copying the newly updated config.json to S3 (with networks to remove)"
+aws s3 cp --acl public-read config.json $BOOTSTRAP_S3_URI
+echo "Done!"
 
-. ./test.common.sh
+sleep 60
+echo "Verifying the networks are being cleaned.."
+node .circleci/testnet-poll-disabled-chains.js
 
-COMMIT_HASH=$(./docker/hash.sh)
+echo "Refreshing config.json and removing the dead networks from it.."
+rm -f config.json && curl -O $BOOTSTRAP_URL
+node .circleci/testnet-remove-disabled-chains.js
 
-# Determine if we have an active PR
-# (The condition means if the variable is not empty)
-if [ ! -z "$CI_PULL_REQUESTS" ]
-then
-    echo "We have an active PR ($CI_PULL_REQUESTS)"
+echo "Copying the newly updated config.json to S3.."
+aws s3 cp --acl public-read config.json $BOOTSTRAP_S3_URI
+echo "Done!"
 
-    echo "Downloading the current testnet Boyar config.json"
-    curl -O $BOOTSTRAP_URL
-    echo "Done downloading! Let's begin by cleaning up the testnet of any stale networks for PRs which already closed"
+echo "Creating a network for this PR within the config.json file.."
+PR_APP_CHAIN_ID=$(node .circleci/testnet-deploy-new-chain-for-pr.js $CI_PULL_REQUESTS $COMMIT_HASH "APP")
+echo "Done adding a new app chain ($PR_APP_CHAIN_ID)"
 
-    node .circleci/testnet-cleanup-mark.js
+echo "Copying the newly updated config.json to S3"
```

Comment:
why are you doing the cleanup and adding a new chain in 3 different steps as opposed to one big change to the config.json file?

Reply:
This is old logic which is here from the get go of chain on pr. It's not new code.
In essence it feels much easier for me to simply issue `curl` and `aws s3 cp` commands to handle the upload and download of the file from S3 instead of writing it to S3. This can be converted to a bash function later on. The actual manipulation on the file is simply easier in Node if we're already discussing the reasoning behind building it like this.
Simply the least amount of code to maintain.

Reply:
@electricmonk - again, old code that was there before.